R/retentionIndexing.R

```diff
@@ -0,0 +1,48 @@
+#' @title Convert retention times to retention indices
```

Comment:
Is that similar to [`MSnbase::alignRt()`](https://github.com/lgatto/MSnbase/blob/655ff179bf149e5abc45fccf4f06a6a6fd34e32a/R/functions-Chromatogram.R#L149-L230)?

I don't know whether this drift is always linear but if not it may be worth to consider some kind of non-linear alignment functions, e.g. [`synapter:::modelRetTime()`](https://github.com/lgatto/synapter/blob/24c0b7b0ae270143bedf16517b6c376a43f22545/R/utils.R#L69-L99) (lowess) or [`MALDIquant::determineWarpingFunctions()`](https://github.com/sgibb/MALDIquant/blob/master/R/determineWarpingFunctions-functions.R) (used for aligning MS1 to MS1 not rtimes but should be easily adoptable, supports lowess, linear, quadratic, cubic).

Reply:
In this case it is not about drift normalization but converting absolute retention times (RTs) to retention indices (RIs), which makes retention data comparable across two different LC systems. Even when using the same column, solvent and gradient differences in absolute RTs are found due to differences like the pumping system, dead volume etc... The conversion from RTs to RIs normalizes for this factors, which allows to use retention data for metabolite identification. This step is often done in GC-MS using the Kovats index.

Reply:
And the files in `inst/retentionIndex` are just for demonstration. One has to first determine the RTs of the indexing substances and then can convert RTs to RIs. The specific example is from an upcoming publication of myself, in which I present an RI system for LC-MS based metabolomics.